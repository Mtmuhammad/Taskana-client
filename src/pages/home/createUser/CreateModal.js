// Modal to create a new project
import React from "react";

const CreateModal = ({ onChange, handleCreate }) => {
  return (
    <div
      className="modal fade"
      id="createUser"
      tabIndex="-1"
      aria-hidden="true"
      data-testid="create-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="createUserLabel"
            >
              {" "}
              Create User
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="firstName" className="form-label">
                First Name
              </label>
              <input
                type="text"
                className="form-control"
                id="firstName"
                name="firstName"
                placeholder="What is the user's first name?"
                onChange={onChange}
                data-testid="input"
              />
            </div>
            <div className="mb-3">
              <label htmlFor="lastName" className="form-label">
                Last Name
              </label>
              <input
                type="text"
                className="form-control"
                id="lastName"
                name="lastName"
                placeholder="What is the user's last name?"
                onChange={onChange}
                data-testid="input"
              />
            </div>

            <div className="mb-3">
              <label htmlFor="email" className="form-label">
                E-mail
              </label>
              <input
                autoComplete="off"
                type="email"
                className="form-control"
                id="email"
                placeholder="Add user email"
                name="email"
                onChange={onChange}
                data-testid="input"
              ></input>
            </div>
            <div className="mb-3">
              <label htmlFor="password" className="form-label">
                Password
              </label>
              <input
                type="password"
                className="form-control"
                id="password"
                placeholder="Add user password"
                name="password"
                onChange={onChange}
                data-testid="input"
                autoComplete="off"
              ></input>
            </div>
            <div className="mb-3">
              <label htmlFor="empRole" className="form-label">
                Employee Role
              </label>
              <select
                onChange={onChange}
                className="form-select"
                id="empRole"
                name="empRole"
              >
                <option value="" defaultValue>
                  Choose a role:
                </option>
                <option value="UI/UX Designer">UI/UX Designer</option>
                <option value="Quality Assurance">Quality Assurance</option>
                <option value="Backend Developer">Backend Developer</option>
                <option value="Web Designer">Web Designer</option>
                <option value="Mobile Developer">Mobile Developer</option>
                <option value="Software Engineer">Software Engineer</option>
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="isAdmin" className="form-label">
                Is this employee admin?
              </label>
              <select
                onChange={onChange}
                className="form-select"
                id="isAdmin"
                name="isAdmin"
              >
                <option value="" defaultValue>
                  Choose a value:
                </option>
                <option value="true">Yes</option>
                <option value="false">No</option>
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              onClick={handleCreate}
              type="button"
              className="fs-5 btn btn-success"
              data-testid="create-btn"
            >
              Create
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CreateModal;
