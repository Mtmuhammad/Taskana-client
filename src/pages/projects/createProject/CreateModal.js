// Modal to create a new project
import React from 'react'

const CreateModal = ({ onChange, handleCreate }) => {
  return (
    <div
      className="modal fade"
      id="createProject"
      tabIndex="-1"
      aria-hidden="true"
      data-testid="create-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5 data-testid="header" className="modal-title  fw-bold" id="createProjectLabel">
              {" "}
              Create Project
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="name" className="form-label">
                Project Name
              </label>
              <input
                type="text"
                className="form-control"
                id="name"
                name="name"
                placeholder="What is the name of this project?"
                onChange={onChange}
                data-testid="input"
              />
            </div>
            <div className="deadline-form">
              <form>
                <div className="row g-3 mb-3">
                  <div className="col">
                    <label htmlFor="deadline" className="form-label">
                      Project End Date
                    </label>
                    <input
                      type="date"
                      className="form-control"
                      id="deadline"
                      name="deadline"
                      onChange={onChange}
                      data-testid="input"
                    />
                  </div>
                </div>
              </form>
            </div>

            <div className="mb-3">
              <label htmlFor="createDescription" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="createDescription"
                rows="3"
                placeholder="Add any extra details about the project"
                name="description"
                onChange={onChange}
                data-testid="input"
              ></textarea>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              onClick={handleCreate}
              type="button"
              className="fs-5 btn btn-success"
              data-testid="create-btn"
            >
              Create
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CreateModal;
