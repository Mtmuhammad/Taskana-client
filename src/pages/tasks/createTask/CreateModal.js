// Modal to create a new task
import React from 'react'

const CreateModal = ({ onChange, handleCreate }) => {
  return (
    <div
      className="modal fade"
      id="createTask"
      tabIndex="-1"
      aria-hidden="true"
      data-testid="create-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="createTaskLabel"
            >
              {" "}
              Create Task
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="name" className="form-label">
                Task Title
              </label>
              <input
                type="text"
                className="form-control"
                id="title"
                name="title"
                placeholder="What is the title of this task?"
                onChange={onChange}
                data-testid="input"
              />
            </div>

            <div className="mb-3">
              <label htmlFor="createTaskDescription" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="createTaskDescription"
                rows="3"
                placeholder="Add any extra details about this task"
                name="description"
                onChange={onChange}
                data-testid="input"
              ></textarea>
            </div>

            <div className="mb-3">
              <label htmlFor="createTaskImportant">Is this task urgent?</label>
              <select
                className="form-control"
                id="createTaskImportant"
                name="important"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
              >
                <option default value="">
                  Choose an option:
                </option>
                <option value="true">Yes</option>
                <option value="false">No</option>
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              onClick={handleCreate}
              type="button"
              className="fs-5 btn btn-success"
              data-testid="create-btn"
            >
              Create
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CreateModal;
