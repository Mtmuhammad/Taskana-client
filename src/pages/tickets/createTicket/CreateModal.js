// Modal to create a new ticket
import React from "react";

const CreateModal = ({ projects, users, onChange, handleCreate }) => {
  return (
    <div
      className="modal fade"
      id="createTicket"
      tabIndex="-1"
      aria-hidden="true"
      data-testid="create-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="createTicketLabel"
            >
              {" "}
              Create Ticket
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="title" className="form-label">
                Ticket Title
              </label>
              <input
                type="text"
                className="form-control"
                id="title"
                name="title"
                placeholder="What is the title of this ticket?"
                onChange={onChange}
                data-testid="input"
              />
            </div>

            <div className="mb-3">
              <label htmlFor="createTicketDescription" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="createTicketDescription"
                rows="3"
                placeholder="Add any extra details about the ticket"
                name="description"
                onChange={onChange}
                data-testid="input"
              ></textarea>
            </div>

            <div className="mb-3">
              <label htmlFor="createTicketAssignedTo">Assigned Name</label>
              <select
                className="form-control"
                id="createTicketAssignedTo"
                name="assignedTo"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
              >
                <option default value="">
                  Choose a team member:
                </option>
                {users &&
                  users.map((user) => (
                    <option value={user?.empNumber}>
                      {user?.firstName} {user?.lastName} ({user.empRole})
                    </option>
                  ))}
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="createTicketProjectId">Project Name</label>
              <select
                className="form-control"
                id="createTicketProjectId"
                name="projectId"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
              >
                <option default value="">
                  Choose a project:
                </option>
                {projects &&
                  projects.map((project) => (
                    <option value={project?.id}>{project?.name}</option>
                  ))}
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              onClick={handleCreate}
              type="button"
              className="fs-5 btn btn-success"
              data-testid="create-btn"
            >
              Create
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CreateModal;
