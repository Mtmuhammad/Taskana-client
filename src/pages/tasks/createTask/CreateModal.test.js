import React from "react";
import { render, screen } from "@testing-library/react";
import CreateModal from "./CreateModal";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<CreateModal />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("create modal", () => {
  test("should render modal", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const modal = screen.getByTestId("create-modal");
    expect(modal).toBeInTheDocument();
    expect(modal).toHaveClass("modal fade");
  });
  test("should render modal header", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const header = screen.getByTestId("header");
    expect(header).toBeInTheDocument();
    expect(header).toHaveClass("modal-title");
    expect(header).toContainHTML("Create Task");
  });
  test("should render close btn (x)", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const close = screen.getByTestId("close-btn");
    expect(close).toBeInTheDocument();
    expect(close).toHaveClass("btn-close");
  });
  test("should render inputs", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const inputs = screen.getAllByTestId("input");
    expect(inputs.length).toBe(3);
  });
  test("should render cancel button", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("cancel-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-secondary");
  });
  test("should render create button", () => {
    render(<CreateModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("create-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-success");
  });
});
