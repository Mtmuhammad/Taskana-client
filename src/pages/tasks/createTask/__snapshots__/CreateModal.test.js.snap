// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="modal fade"
    data-testid="create-modal"
    id="createTask"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title  fw-bold"
            data-testid="header"
            id="createProjectLabel"
          >
             Create Task
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            data-testid="close-btn"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="name"
            >
              Task Title
            </label>
            <input
              class="form-control"
              data-testid="input"
              id="title"
              name="title"
              placeholder="What is the title of this task?"
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="createTaskDescription"
            >
              Description (optional)
            </label>
            <textarea
              class="form-control"
              data-testid="input"
              id="createTaskDescription"
              name="description"
              placeholder="Add any extra details about this task"
              rows="3"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="createTaskImportant"
            >
              Is this task urgent?
            </label>
            <select
              class="form-control"
              data-testid="input"
              id="createTaskImportant"
              name="important"
              style="font-size: 1.3rem;"
            >
              <option
                default=""
                value=""
              >
                Choose an option:
              </option>
              <option
                value="true"
              >
                Yes
              </option>
              <option
                value="false"
              >
                No
              </option>
            </select>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="fs-5 btn btn-secondary"
            data-bs-dismiss="modal"
            data-testid="cancel-btn"
            type="button"
          >
            Cancel
          </button>
          <button
            class="fs-5 btn btn-success"
            data-bs-dismiss="modal"
            data-testid="create-btn"
            type="button"
          >
            Create
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
